### ISSUE TITLE
Create build and deploy GitHub actions

### ISSUE THREAD

**Issue by ScottLilly** [ScottLilly/ArgumentParser#5]

**Comment by ScottLilly** [comment]
Half of this is done. The other half is blocked on a NuGet.org API key, which has to be created by hand.

## Done: build and release

`.github/workflows/release.yml`, added in 7b2b3e6. Manually triggered from the Actions tab, it builds, tests, packs, and creates a GitHub Release with the `.nupkg` and `.snupkg` attached.

The version is read back off the packed file, so the release tag can never disagree with the package. Bump `<Version>` in `ArgumentParser/ArgumentParser.csproj` and the tag follows. The run stops before creating anything if that tag already exists, so the current `1.1.1.0` will be refused until it is bumped.

`ci.yml` still handles build and test on every push and pull request to `master`.

## Remaining: push to NuGet.org

One extra step in the same workflow, once the key exists.

**1. Create the API key on NuGet.org**

Sign in at nuget.org, then account menu -> API Keys -> Create.

| Field | Value |
|---|---|
| Key Name | Something identifying it, e.g. `ArgumentParser GitHub Actions` |
| Scopes | `Push` only. Do not grant `Unlist` |
| Select Packages | `ScottLilly.ArgumentParser`, or the glob `ScottLilly.*` to cover the other packages with one key |
| Expiration | 365 days is the maximum NuGet allows |

Scoping the key to the one package limits the damage if it leaks. The key is shown once, on creation, and cannot be retrieved afterward.

**2. Add it as a GitHub repository secret**

Repo Settings -> Secrets and variables -> Actions -> New repository secret.

| Field | Value |
|---|---|
| Name | `NUGET_API_KEY` |
| Secret | The key from step 1 |

The workflow will read it as `${{ secrets.NUGET_API_KEY }}`. Secrets are not exposed to workflows triggered from forks, which is fine here because this one is manually triggered.

**3. Decide whether to gate the push**

Optional. A GitHub Environment (Settings -> Environments) with a required reviewer would make the push step wait for an explicit approval, on top of the workflow already being manual. Worth it because of the next point.

## Worth knowing before turning this on

**A push to NuGet.org cannot be undone.** A published version cannot be deleted, only unlisted, and unlisting does not stop anyone who already references that exact version from restoring it. The version number is consumed permanently, so a mistake means burning a version number and publishing a corrected one. This repo already has an unlisted 1.1.0, which is what that looks like in practice.

For that reason the push should use `--skip-duplicate`, so re-running the workflow against an already-published version is a no-op rather than an error.

**The symbol package goes up on its own.** `dotnet nuget push` sends the matching `.snupkg` alongside the `.nupkg` when both are in the same folder, which they are. No second command needed.

## The expiration will bite

The key stops working after at most a year, and the failure arrives as a `403` in the middle of a release. Setting a calendar reminder to rotate it is worth more than it sounds.

The same key and the same remaining step apply to ScottLilly/CSharpExtender#3 and to FluentEmail, so one `ScottLilly.*` scoped key would cover all three.

**Comment by ScottLilly** [comment]
Closing. Both actions exist and both work, which is what the title asked for.

- `ci.yml` builds and tests on every push to `master` and on pull requests against it, restored in dfa6861 after a spell as manual-only. The first automatic run passed on a clean Ubuntu runner: the multi-targeted build and all 187 tests, in 33 seconds. The Build Status badge in the README now tracks `master` rather than whichever manual run happened last.
- `release.yml` builds, tests, packs, and creates a GitHub Release with the `.nupkg` and `.snupkg` attached. Manually triggered, deliberately, since it publishes something.

**Publishing to NuGet.org is not automated, and is not being automated here.** That was the other half described in the comment above, and it is blocked on an API key that has to be created by hand. Packing and releasing are automated; pushing the package to nuget.org stays a manual step.

The setup instructions for that remaining step are in the comment above and stay readable on this closed issue. They are also referenced from ScottLilly/CSharpExtender#3, and the same `ScottLilly.*` scoped key would cover all three packages, so nothing is lost by closing this one.

The `deferred` label is removed, since this is no longer waiting on a decision.
